test/dynamo/test_fake_distributed.py

```diff
@@ -0,0 +1,121 @@
+# Owner(s): ["module: dynamo"]
+from unittest import skipIf
+
+import torch
+import torch.distributed as dist
+from torch._dynamo.test_case import TestCase as DynamoTestCase
+from torch._dynamo.testing import AotEagerAndRecordGraphs, normalize_gm
+from torch.distributed._functional_collectives import (
+    all_to_all_single_autograd,
+    wait_tensor,
+)
+from torch.testing._internal.common_utils import instantiate_parametrized_tests
+from torch.testing._internal.distributed.fake_pg import FakeStore
+
+
+def normalize_graph(gm):
+    return normalize_gm(gm.print_readable(print_output=False))
+
+
+@skipIf(not dist.is_available(), "requires distributed")
+class TestFakeDistributed(DynamoTestCase):
+    def setUp(self):
+        # Use FakeProcessGroup to run tests on a single process
+        self.store = FakeStore()
+        dist.init_process_group(backend="fake", rank=0, world_size=2, store=self.store)
+
+    def tearDown(self):
+        dist.destroy_process_group()
+
+    def test_all_to_all_single_autograd(self):
+        backend = AotEagerAndRecordGraphs()
+
+        @torch.compile(fullgraph=True, backend=backend)
+        def fn(x):
+            return all_to_all_single_autograd(
+                x,
+                None,  # Will use equal splits
+                None,  # Will use equal splits
+                group=dist.group.WORLD,
+            )
+
+        # Test backed shapes
+        x = torch.randn(8, 8, requires_grad=True)
+        torch._dynamo.mark_dynamic(x, 0)
+        torch._dynamo.mark_dynamic(x, 1)
+        wait_tensor(fn(x))
+        self.assertEqual(len(backend.fw_graphs), 1)
+        self.assertEqual(len(backend.bw_graphs), 1)
+        self.assertExpectedInline(
+            normalize_graph(backend.fw_graphs[0]),
+            """\
+class GraphModule(torch.nn.Module):
+    def forward(self, primals_1: "Sym(s77)", primals_2: "Sym(s27)", primals_3: "f32[s77, s27]"):
+        floordiv: "Sym((s77//2))" = primals_1 // 2
```

Comment:
Just curious, does the normalize helper ensure these symint numberings are stable? Seems to me if someone adds another symint supported op that's used in this test case it would destabilize the numberings, though that's unlikely since the test case is just one op